addons/pos_cash_rounding/static/src/js/pos_cash_rounding.js

```diff
@@ -43,7 +43,9 @@ models.Order = models.Order.extend({
 
             var rounding_applied = total - this.get_total_with_tax();
             // because floor and ceil doesn't include decimals in calculation, we reuse the value of the half-up and adapt it.
-            if(this.pos.cash_rounding[0].rounding_method === "UP" && rounding_applied < 0) {
+            if (utils.float_is_zero(rounding_applied)){
+                // https://xkcd.com/217/
```

Comment:
This would be a bit clearer I think:
```suggestion
                // https://xkcd.com/217/
                return 0;
```